File: Codeshef/Long/six.py
R=raw_input

n,m = map(int, R().split())

arr = []

for i in range(n):
	arr.append(map(int, R().split()))

q = int(R())

for qq in range(q):
	a,b = map(int,R().split())
	temp = float('inf')
	for i in range(n):
		for j in range(m):
			if i+a>n or j+b>m:
				continue
			maxx = -1*float('inf')
			summ = 0
			for x in range(a):
				for y in range(b):
					maxx = max(maxx, arr[i+x][j+y])
					summ += arr[i+x][j+y]
			temp = min(temp, maxx*a*b - summ)
	print temp	

File: Codeshef/Long/six.cpp
#include <bits/stdc++.h>
#include <cmath>
using namespace std;

int main(int argc, char const *argv[]) {
	int n,m;
	cin>>n>>m;

	int arr[n][m];

	for(int i=0; i<n; ++i)
		for(int j=0; j<m ;++j)
			cin>>arr[i][j];

	int q; 
	cin>>q;
	
	int a,b;
	long long ans, maxx, sum, temp;
	while(q--){
		cin>>a>>b;
		ans = INT_MAX;
		for(int i=0; i<n; ++i){
			for(int j=0; j<m ; ++j){
				if(i+a>n || j+b>m) continue;

				maxx = INT_MIN;
				sum = 0;
				for(int x = 0; x<a; ++x){
					for(int y = 0; y<b; ++y){
						maxx = max(maxx, arr[i+x][j+y]);
						sum+=arr[i+x][j+y];
					}
				}
				temp = maxx*a*b - sum;
				ans = min(ans,temp);
				// cout<<"maxx : "<<maxx<<" ans : "<<ans<<endl;
			}
		}
		cout<<ans<<endl;
	}
	return 0;
}